core-api/src/main/java/com/sequenceiq/cloudbreak/validation/UrlValidator.java

```diff
@@ -0,0 +1,32 @@
+package com.sequenceiq.cloudbreak.validation;
+
+import static org.apache.commons.validator.routines.UrlValidator.ALLOW_ALL_SCHEMES;
+import static org.apache.commons.validator.routines.UrlValidator.ALLOW_LOCAL_URLS;
+import static org.apache.commons.validator.routines.UrlValidator.NO_FRAGMENTS;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class UrlValidator implements ConstraintValidator<ValidUrl, String> {
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        context.disableDefaultConstraintViolation();
+        if (value == null) {
+            return true;
+        }
+        if (value.isEmpty()) {
+            ValidatorUtil.addConstraintViolation(context, "URL may not be empty");
+            return false;
+        }
+
+        long validatorOptions = ALLOW_ALL_SCHEMES + ALLOW_LOCAL_URLS + NO_FRAGMENTS;
```

Comment:
Actually I would make it as strict as possible, but this is a nuance thing so might not be worth it.